src/tests/reducers/user.test.js

```diff
@@ -39,6 +40,26 @@ describe('User reducer', () => {
             .toEqual({ ...state, profile: user });
         expect(window.localStorage.setItem).toHaveBeenCalledWith("profile", JSON.stringify(user));
     });
+
+    it('Clear user', () => {
+        Object.defineProperty(window,
+            "localStorage",
+            {
+                value: {
+                    getItem: jest.fn(key => `{"key":"${key}"}`),
+                    setItem: jest.fn((key, value) => { }),
+                    removeItem: jest.fn(),
+                    clear: jest.fn(),
+                },
+            });
+
+        const state = userInitialState();
+       
+        expect(userReducer(state, setUser()))
+            .toEqual({ ...state, profile: undefined, auth: undefined });
+        expect(window.localStorage.clear).toHaveBeenCalled();
```

Comment:
🔴 **Нужно исправить:**

Текущий тест падает с ошибкой  (см. https://disk.yandex.ru/i/BpWIjB4tHEdEEQ)
Для исправления данной ситуации имеет смысл поправить мок `localStorage`.

В качестве одного из хороших вариантов таких исправлений является использование `spyOn`:

```js

 let setItemSpy, getItemSpy, resetSpy;
  const mockStorage = {};

  beforeAll(() => {
    setItemSpy = jest
      .spyOn(global.Storage.prototype, "setItem")
      .mockImplementation((key, value) => {
        mockStorage[key] = value;
      });

    getItemSpy = jest
      .spyOn(global.Storage.prototype, "getItem")
      .mockImplementation((key) => mockStorage[key]);

    resetSpy = jest
      .spyOn(global.Storage.prototype, "clear")
      .mockImplementation(() =>
        Object.keys(mockStorage).forEach((key) => delete mockStorage[key])
      );
  });

  afterAll(() => {
    // then, detach our spies to avoid breaking other test suites
    getItemSpy.mockRestore();
    setItemSpy.mockRestore();
    resetSpy.mockRestore();
  });

```

Это позволит вам не переопределять `localStorage` в каждом из тестов и можно будет обращаться без `window` к данному хранилищу: `expect(localStorage.clear).toHaveBeenCalled()`